models/odos/avalanche_c/odos_avalanche_c_trades.sql

```diff
@@ -0,0 +1,122 @@
+{{ config(
+    alias = 'trades',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address'],
+    post_hook='{{ expose_spells(\'["avalanche_c"]\',
+                                "project",
+                                "odos",
+                                \'["Henrystats"]\') }}'
+    )
+}}
+
+
+{% set project_start_date = '2022-11-29' %}
+
+WITH 
+
+dexs_raw as (
+        SELECT 
+            evt_block_time as block_time, 
+            explode(outputs) as data_value, 
+            '' as maker, 
+            CAST(CONCAT_WS(", ", amountsIn) as double) as token_sold_amount_raw, 
```

Comment:
i don't know the answer off the top of my head, but with dex aggregator spell, are we enforcing one row per trade, that's why we concat values here? if so, will it still be simple to aggregate in queries for dashboards downstream? this may be a larger conversation, with this spell being so new relative to dex trades. it may be an opportunity to leverage the new GH discussions page to talk about how the community would like to see final product of `dex_aggregator.trades` -- what do you think?

Reply:
nope, there's only one trade in a row. We're not concating the values, they're stored in an array(string) and that was the only method to get them.

Reply:
something about the usage of `concat_ws()` function here isn't sitting right with me. if the column is an array & you can ensure that there'll only ever be one value, you can simply do `amountsIn[0]` to select the value out of the array, then cast the datatype. this is applicable to all use cases of `concat_ws()` in this spell. can you play with that and let me know your thoughts? i just did a quick query and it looks to work fine

Reply:
uhm weird, I'm trying to remember why I had to use concat_ws 😅. just pushed an update